contrib/platform/test/com/sun/jna/platform/win32/WevtapiTest.java

```diff
@@ -0,0 +1,574 @@
+            assertThat(((WinDef.BOOL) evtVariant.getValue()).booleanValue(), is(true));
+        } finally {
+            if (channelHandle != null) {
+                Wevtapi.INSTANCE.EvtClose(channelHandle);
+            }
+        }
+    }
+
+    public void testModifyChannelConfig() throws Exception {
+        EVT_HANDLE channelHandle = null;
+        try {
+            channelHandle = Wevtapi.INSTANCE.EvtOpenChannelConfig(null, "Application", 0);
+            assertNotNull(channelHandle);
+
+            Winevt.EVT_VARIANT evtVariant = WevtapiUtil.EvtGetChannelConfigProperty(channelHandle, EVT_CHANNEL_CONFIG_PROPERTY_ID.EvtChannelConfigClassicEventlog);
+            assertThat(((WinDef.BOOL) evtVariant.getValue()).booleanValue(), is(true));
+
+            Winevt.EVT_VARIANT setter = new Winevt.EVT_VARIANT();
+            setter.setValue(Winevt.EVT_VARIANT_TYPE.EvtVarTypeBoolean, new BOOL(false));
+            Wevtapi.INSTANCE.EvtSetChannelConfigProperty(channelHandle, EVT_CHANNEL_CONFIG_PROPERTY_ID.EvtChannelConfigClassicEventlog, 0, setter);
+
+            evtVariant = WevtapiUtil.EvtGetChannelConfigProperty(channelHandle, EVT_CHANNEL_CONFIG_PROPERTY_ID.EvtChannelConfigClassicEventlog);
+            assertThat(((WinDef.BOOL) evtVariant.getValue()).booleanValue(), is(false));
+
+            setter.setValue(Winevt.EVT_VARIANT_TYPE.EvtVarTypeBoolean, new BOOL(true));
+            Wevtapi.INSTANCE.EvtSetChannelConfigProperty(channelHandle, EVT_CHANNEL_CONFIG_PROPERTY_ID.EvtChannelConfigClassicEventlog, 0, setter);
+
+            evtVariant = WevtapiUtil.EvtGetChannelConfigProperty(channelHandle, EVT_CHANNEL_CONFIG_PROPERTY_ID.EvtChannelConfigClassicEventlog);
+            assertThat(((WinDef.BOOL) evtVariant.getValue()).booleanValue(), is(true));
+
+            // Writing back is skipped neighter is EvtChannelConfigClassicEventlog
+            // writable, nor is it a good idea to mess with the log of the developer machine
+        } finally {
+            if (channelHandle != null) {
+                Wevtapi.INSTANCE.EvtClose(channelHandle);
+            }
+        }
+    }
+
+    public void testEvtOpenPublisherEnum() throws Exception {
+        Winevt.EVT_RPC_LOGIN login = new Winevt.EVT_RPC_LOGIN("localhost", null, null, null,
+                Winevt.EVT_RPC_LOGIN_FLAGS.EvtRpcLoginAuthDefault);
+        EVT_HANDLE session = null;
+        EVT_HANDLE publisherEnumHandle = null;
+        List<String> publisherList = new ArrayList<String>();
+        try {
+            session = Wevtapi.INSTANCE.EvtOpenSession(Winevt.EVT_LOGIN_CLASS.EvtRpcLogin, login, 0, 0);
+            if (session == null) {
+                throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+            }
+            publisherEnumHandle = Wevtapi.INSTANCE.EvtOpenPublisherEnum(session, 0);
+            if (publisherEnumHandle == null) {
+                throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+            }
+
+            Memory buff;

```

Comment:
With the suggested changes to WevtapiUtil you'll get a String back directly, so in that case this needs adjusting.